### DISCUSSION TITLE
Detect TablePlus and Tinkerwell in non-standard locations

### DISCUSSION THREAD

**Discussion by Sagmedjo** [beyondcode/herd-community#945] [Ideas]
### Platform

Windows

### Operating system version

Windows 11 Pro Version	23H2

### System architecture

Windows

### Herd Version

1.10.0

### Bug description

# Prerequisites

TablePlus and Tinkerwell already installed with latest versions on my Machine, but not in the default directories.

# Bug

Before and after updating to 1.10.0 Herd always opens the command line instead of Tinkerwell.
After updating to 1.10.0 Herd still opens up Adminer instead of TablePlus

**Comment by sschlein** [comment]
Does it work if you move them to the default directories?

**Comment by Sagmedjo** [comment]
I actually do not want to move them to the default directories because of my file and folder structure. All my new installations are made to my "D" drive because the "C" drive is reserved for files and other data. All applications which are still on my "C" drive should also be moved to my "D" drive one after another.

I really do think that the detection should not depend on the folder location. But if it helps to understand the bug. These are the installation paths for both applications:

- Tinkerwell
  * C:\Users\user\AppData\Local\Programs\Tinkerwell
- TablePlus
  * D:\Programs\TablePlus

**Comment by sschlein** [comment]
I am moving this to a discussion because that's an idea for an improvement and not a bug. Most of the time, we focus on work that improves Herd for at least 1% of the user base, so if there are enough requests for this, we'll implement a detection that is beyond the current version.

**Comment by isimmons** [comment]
I get the same issue on Windows 11 with 1.10 and 1.11 Herd and both tinkerwell and tableplus are in the default directories (assuming this means where they install by default when I install them)

`Users\isimm\AppData\Local\Programs\Tinkerwell\Tinkerwell.exe` and `"C:\Program Files\TablePlus\TablePlus.exe"`

Clicking open for tableplus opens adminer in the browser and open for tinkerl opens powershell

**Comment by charlielangridge** [comment]
I also have the issue where Herd doesnt detect Tinkerwell. It finds table plus, but only links to it from the Database service, not the sites page
